nix/overlays/checked-shell-script/checked-shell-script.nix

```diff
@@ -1,17 +1,48 @@
 # Create a bash script that is checked with shellcheck. You can either use it
 # directly, or use the .bin attribute to get the script in a bin/ directory,
 # to be used in a path for example.
-{ git
+{ argbash
+, bash_5
+, git
 , lib
 , runCommand
-, runtimeShell
 , shellcheck
 , stdenv
 , writeTextFile
 }:
-{ name, docs, inRootDir ? false, redirectTixFiles ? true, withTmpDir ? false }: text:
-# TODO: do something sensible with docs, e.g. provide automated --help
+{ name
+, docs
+, args ? [ ]
+, inRootDir ? false
+, redirectTixFiles ? true
+, withEnv ? null
+, withTmpDir ? false
+}: text:
 let
+  argsParser =
+    writeTextFile {
+      name = "${name}-args";
+      destination = "/parser.sh";
+
+      text =
+        ''
+          # BASH_ARGV0 sets $0 - which is used in parser.sh for usage information
+          # stripping the /nix/store/... path for nicer display
+          BASH_ARGV0="$(basename "$0")"
+
+          # ARG_HELP([${name}], [${lib.strings.stringAsChars (c: if c == "\n" then " \\n" else c) docs}])
```

Comment:
I think we'd need to escape `[` and `]`, too, to not make the m4 macro fail. Short of adding some crazy nested `if`s... any other utility function I could use for that?

Reply:
builtins.stringReplace I think